src/NServiceBus.Transport.RabbitMQ.TransportTests/ConfigureRabbitMQTransportInfrastructure.cs

```diff
@@ -55,7 +46,38 @@ public Task Cleanup()
 
     void PurgeQueues()
     {
-        var connectionFactory = CreateConnectionFactory();
+        if (connectionStringBuilder == null)
+        {
+            return;
+        }
+
+        var connectionFactory = new ConnectionFactory
+        {
+            AutomaticRecoveryEnabled = true,
+            UseBackgroundThreadsForIO = true
+        };
+
+        object value;
+
+        if (connectionStringBuilder.TryGetValue("username", out value))
+        {
+            connectionFactory.UserName = value?.ToString();
+        }
+
+        if (connectionStringBuilder.TryGetValue("password", out value))
+        {
+            connectionFactory.Password = value?.ToString();
+        }
+
+        if (connectionStringBuilder.TryGetValue("virtualhost", out value))
+        {
+            connectionFactory.VirtualHost = value?.ToString();
+        }
+
+        if (connectionStringBuilder.TryGetValue("host", out value))
+        {
+            connectionFactory.HostName = value?.ToString();
+        }
```

Comment:
For consistency in how we handle the default values for the transport, it seems like we should throw if we don't have a hosthame, since that would already be considered an incorrect connection string by the transport.